frontend/src/components/WindSettings.jsx

```diff
@@ -19,390 +19,153 @@ const useStyles = makeStyles((theme) => ({
-                        {/* {windShears.map((shear, index) => ( 
-                                <Grid item xs={12} key={index}>
-                                    <div style={{ display: 'flex', alignItems: 'center' }}>
-                                        <ul style={{ flex: '1', marginRight: '10px' }}>
-                                            <li>
-                                                Direction: {shear.windDirection}, Velocity: {shear.windVelocity}, Fluctuation %: {shear.fluctuationPercentage}
-                                            </li>
-                                        </ul>
-                                        <IconButton onClick={() => deleteWindShear(index)} color="blue">
-                                            <CloseIcon />
-                                        </IconButton>
-                                    </div>
-                                </Grid>
-                        ))}
+    const renderTextField = (label, value, onChange, inputProps) => (
+        <Grid item container alignItems="center" direction="row">
+            <Grid item xs={4}>
+                <InputLabel sx={{ marginRight: 2, flexShrink: 0, color: '#F5F5DC', width: '200px' }}>{label}</InputLabel>
+            </Grid>
+            <Grid item xs={6}>
+                <Tooltip title={`Enter ${label}`} placement='bottom'>
+                    <TextField
+                        sx={{
+                            backgroundColor: '#F5F5DC',
+                            '& .MuiOutlinedInput-root': {
+                                '& .MuiInputBase-input': {
+                                    padding: '6px 8px',
+                                },
+                            },
+                        }}
+                        variant="outlined"
+                        type="number"
+                        onChange={onChange}
+                        value={value}
+                        inputProps={inputProps}
+                        fullWidth
+                    />
+                </Tooltip>
+            </Grid>
+        </Grid>
+    );
+
+    const addNewWindBlock = () => {
+        const newWindBlock = {
+            windType: selectedWindType,
+            windDirection: envConf.Wind.Direction,
+            windVelocity: envConf.Wind.Force,
+            fluctuationPercentage: selectedWindType === "Turbulent Wind" ? fluctuationPercentage : 0,
+        };
+        setWindBlocks([...windBlocks, newWindBlock]);
+    };
+
+    setWindBlockData = (index, updatedData) => {
+        const updatedWindBlocks = [...windBlocks];
+        updatedWindBlocks[index] = { ...updatedWindBlocks[index], ...updatedData };
+        setWindBlocks(updatedWindBlocks);
+    };
+
+    deleteWindBlock = (index) => {
+        const updatedWindBlocks = [...windBlocks];
+        updatedWindBlocks.splice(index, 1);
+        setWindBlocks(updatedWindBlocks);
+    };
 
```

Comment:
@syedomair0 Visually everything looks good to me. 
One issue is found in its functionality. Let me explain the failed test case:
 I created 3 wind blocks. When I attempted to delete 2nd block, the 3rd block has been deleted instead of 2nd.
The same occurs when I attempted to delete 3rd block from the list of 4 blocks. 4th block has been deleted instead of 3rd. 
Print the windBlocks array to console and debug this issue.

Reply:
thank you for pointing this out. debugged and fixed.